### ISSUE TITLE
request: warn on simple infinite recursion

### ISSUE THREAD

**Issue by mcandre** [koalaman/shellcheck#2742]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
sh
```

(Or calling `bash` without arguments from a `bash` script, or `ash` from an `ash` script, ... and so on ..., or `zsh` from a `zsh` script.)

Frankly, invoking *any* shell interpreter from any other shell interpreter, with no other CLI arguments, is likely to be a mistake.

#### Here's what shellcheck currently says:

N/A

#### Here's what I wanted or expected to see:

Warning: Risk of infinite recursion. Supply CLI argument(s) to `sh`.
